OMCompiler/SimulationRuntime/c/simulation/solver/nonlinearSystem.c

```diff
@@ -418,6 +405,23 @@ int initializeNonlinearSystems(DATA *data, threadData_t *threadData)
     nonlinsys[i].max = (double*) malloc(size*sizeof(double));
     nonlinsys[i].initializeStaticNLSData(data, threadData, &nonlinsys[i]);
 
+#if !defined(OMC_MINIMAL_RUNTIME)
+    // phannebohm:
+    // this can only be done AFTER initializeStaticNLSData, because isPatternAvailable is set there.
+    // how did this even work before?!?
+    if (nonlinsys[i].isPatternAvailable)
```

Comment:
I guess it worked only by luck. 
Or does `omc_alloc_interface.malloc_uncollectable` initialize everything with zeros?

```C
data->simulationInfo->nonlinearSystemData = (NONLINEAR_SYSTEM_DATA*) omc_alloc_interface.malloc_uncollectable(data->modelData->nNonLinearSystems*sizeof(NONLINEAR_SYSTEM_DATA));
```

Reply:
But I would remove the comment It is strange how it was working before, but you could have the reason for the change in the commit message.

Reply:
It probably initializes with zeros. But that's exactly the problem! If `isPatternAvailable` is not set properly, how can we ever detect to use the sparse solver at this place?

Reply:
Maybe it was done somewhere again else? We should probably check that, since I was pretty sure we are using the sparsity pattern at some places. Do you have a test case?